src/topp/FileInfo.cpp

```diff
@@ -376,19 +404,73 @@ class TOPPFileInfo :
     // Content statistics
     //-------------------------------------------------------------
     Map<String, int> meta_names;
-    if (in_type == FileTypes::FEATUREXML) //features
+
+	if (in_type == FileTypes::FASTA)
+	{
+		vector <FASTAFile::FASTAEntry> entries;
+		FASTAFile file;
+		vector <FASTAFile::FASTAEntry>::iterator loopiter;
+		vector <FASTAFile::FASTAEntry>::iterator iter;
+
+		map<char, int> aacids;
+		std::map<char, int>::iterator it;
+		int number_of_aacids = 0;
+		
+		mu.before();
+		//loading input
+		file.load(in, entries);
+
+		mu.after();
+		std::cout << "\n\nMemory usage while loading: " << mu << std::endl;
+
+		os << "Number of sequences: " << entries.size() << "\n"
+			<< "\n";
+	
+		for (loopiter = entries.begin(); loopiter != entries.end(); loopiter=std::next(loopiter))
+		{
+			iter = find_if(entries.begin(), loopiter, bind1st(mem_fun(&FASTAFile::FASTAEntry::headerMatches), &(*loopiter)));
+
+			if (iter != loopiter)
+			{
+				os << "Warning: Duplicate sequence, Number: " << std::distance(entries.begin(), loopiter) << ", ID: " << loopiter->identifier << " is same as Number: " << std::distance(entries.begin(), iter) << ", ID: " << iter->identifier << "\n";
+			}
+
+			iter = find_if(entries.begin(), loopiter, bind1st(mem_fun(&FASTAFile::FASTAEntry::sequenceMatches), &(*loopiter)));
+		    
+			if (iter != loopiter && iter != entries.end())
+			{
+				os << "Warning: Duplicate header, Number: "<< std::distance(entries.begin(), loopiter) << ", ID: " << loopiter->identifier <<" is same as Number: " << std::distance(entries.begin(), iter) << ", ID: " << iter->identifier << "\n";
+			}
+
+			for (int i = 0; i < loopiter->sequence.size(); i++)
+			{
+				++aacids[loopiter->sequence[i]];
+			}
+			number_of_aacids += loopiter->sequence.size();
+		}
+
+		
+		os << "Total amino acids: " << number_of_aacids << "\n\n";
+		os << "Amino acid counts: " << "\n";
+
+		for (it = aacids.begin(); it != aacids.end(); it=std::next(it))
+		{
+			os << it->first << '\t' << it->second << "\n\n";
```

Comment:
Yes I think so too, so changing it.
Do you think the test file should have more than one sequence, and with repetition?